DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/part1_pointers/src/mmhost.cpp

```diff
@@ -0,0 +1,89 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+struct PointerIP {
+  // Pointer kernel arguments will be passed through the component's CSR. They
+  // will refer to data accessible through a shared Avalon memory-mapped host
+  // interface.
+  int *x;
+  int *y;
+  int *z;
+  int size;
+
+  void operator()() const {
+    for (int i = 0; i < size; ++i) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+
+int main(void) {
+#if FPGA_SIMULATOR
+  auto selector = sycl::ext::intel::fpga_simulator_selector_v;
+#elif FPGA_HARDWARE
+  auto selector = sycl::ext::intel::fpga_selector_v;
+#else  // #if FPGA_EMULATOR
+  auto selector = sycl::ext::intel::fpga_emulator_selector_v;
+#endif
+
+  bool passed = true;
+
+  try {
+    // create the device queue
+    sycl::queue q(selector, fpga_tools::exception_handler,
+                  sycl::property::queue::enable_profiling{});
+
+    // Print out the device information.
+    sycl::device device = q.get_device();
+    std::cout << "Running on device: "
+              << q.get_device().get_info<sycl::info::device::name>().c_str()
+              << std::endl;
+
+    // Create and initialize the host arrays
+    constexpr int kN = 8;
+    std::cout << "Elements in vector : " << kN << "\n";
+
+    // Here we may use auto* or int* when declaring the pointer interface
+    auto *array_A = sycl::malloc_shared<int>(kN, q);
+    auto *array_B = sycl::malloc_shared<int>(kN, q);
+    int *array_C = sycl::malloc_shared<int>(kN, q);
```

Comment:
```suggestion
    int *array_a = sycl::malloc_shared<int>(kN, q);
    int *array_b = sycl::malloc_shared<int>(kN, q);
    int *array_c = sycl::malloc_shared<int>(kN, q);
```
I don't see the value of telling users they can use `auto` - this is unrelated to this sample right?

Variables should be snake_case